docs/extreme.md

```diff
@@ -37,36 +43,52 @@ This has a couple of important caveats:
   signals. If there are more handlers registered than just our own, we will
   simply flush the extreme mode buffer.
 
-So in summary, only use extreme mode when performing an extreme amount of
-logging and it is acceptable to potentially lose the most recent logs.
-
 ## Usage
 
-Extreme mode is defined by creating an extreme mode destination with
-`pino.extreme()`.
-
-The following creates an extreme destination to stdout:
+The `pino.extreme()` method will provide an Extreme Mode destination.
 
 ```js
-'use strict'
+const pino = require('pino')
+const dest = pino.extreme() // logs to stdout with no args
+const logger = pino(dest)
+```
 
+## Log loss prevention
+
+The following strategy can be used to minimize log loss:
+
+```js
 const pino = require('pino')
 const dest = pino.extreme() // no arguments
 const logger = pino(dest)
 
+// flush every 10 seconds to keep the buffer empty 
+// in periods of low activity
 setInterval(function () {
-  // flush is asynchronous
-  dest.flush()
+  logger.flush()
 }, 10000).unref()
+
+// use pino.final to create a special logger that 
+// guarantees final tick writes 
+const handler = pino.final(logger, (err, finalLogger, evt) => {
+  if (err) finalLogger.error(err, 'error caused exit')
+  finalLogger.info(`${evt} caught`)
+})
+// catch all the ways node might exit
+process.on('beforeExit', () => handler(null, 'beforeExit'))
+process.on('exit', () => handler(null, 'exit'))
+process.on('uncaughtException', (err) => handler(err, 'uncaughtException'))
+process.on('SIGHUP', () => handler(null, 'SIGHUP'))
+process.on('SIGINT', () => handler(null, 'SIGINT'))
+process.on('SIGQUIT', () => handler(null, 'SIGQUIT'))
+process.on('SIGTERM', () => handler(null, 'SIGTERM'))
```

Comment:
This example is not totally correct. On some of those event we would like to exit with an error code (or not), in some other we want to just flush.

Specifically with `SIGHUP`, we were just flushing asynchronously. I think that behavior should be documented as well.

Reply:
I'll remove SIGHUP if it's a special case, we can add something to the help section about it